deeppavlov/models/ranking/matching_models/tf_base_matching_model.py

```diff
@@ -0,0 +1,158 @@
+"""
+Copyright 2018 Neural Networks and Deep Learning lab, MIPT
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import numpy as np
+from typing import List, Iterable, Union
+
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.models.ranking.siamese_model import SiameseModel
+from deeppavlov.core.models.tf_model import TFModel
+
+log = get_logger(__name__)
+
+
+class TensorflowBaseMatchingModel(TFModel, SiameseModel):
```

Comment:
If this is an abstract class, it should be marked as such
```suggestion
class TensorflowBaseMatchingModel(TFModel, SiameseModel, metaclass=ABC):
```